src/core/Oscillator.cpp

```diff
@@ -88,6 +118,206 @@ void Oscillator::update( sampleFrame * _ab, const fpp_t _frames,
 	}
 }
 
+void Oscillator::generateSineWaveTable(int bands)
+{
+	bands = bands;
+	m_generatedWaveTable = new sample_t[WAVETABLE_LENGTH];
+	for (int i = 0; i < WAVETABLE_LENGTH; i++)
+	{
+		m_generatedWaveTable[i] = sinf(((float)i / (float)WAVETABLE_LENGTH) * F_2PI);
+	}
+}
+
+void Oscillator::generateSawWaveTable(int bands)
+{
+	float max = 0;
+	m_generatedWaveTable = new sample_t[WAVETABLE_LENGTH];
```

Comment:
These `generate<X>WaveTable()` functions all leak the allocation associated with `m_generatedWaveTable` (i.e. there's no `delete` matching this `new` statement). I would advise just returning a `std::array<sample_t, WAVETABLE_LENGTH>` type rather than returning the data through a heap-allocated member variable. i.e.
```
std::array<sample_t, WAVETABLE_LENGTH> Oscillator::generateSawWaveTable(int bands)
{
    std::array<sample_t, WAVETABLE_LENGTH> wave;
    // populate wave[i] for each index i.
    return wave;
}
```

Reply:
A few more alternatives:
- Pass the address to write results
- Use arrays and/or some smart pointers for `s_waveTables`

Reply:
Also, I think you may reduce duplications by using passing waveform type/pointer to a function that returns coefficient.

Reply:
The generation functions have been revised to take a pointer to the relevant location in `s_waveTables`. This addresses the issue of the memory leakage, and removes unnecessarily verbose code.